File: web/src/plugin/tools/api/userRelation.js
import service from '@/utils/request'

export const createUserRelation = (data) => {
  return service({
    url: '/toolsUserRelation/createUserRelation',
    method: 'post',
    data
  })
}

export const findUserRelation = (params) => {
  return service({
    url: '/toolsUserRelation/findUserRelation',
    method: 'get',
    params
  })
}

export const getUserRelationList = (params) => {
  return service({
    url: '/toolsUserRelation/getUserRelationList',
    method: 'get',
    params
  })
}

export const deleteUserRelation = (params) => {
  return service({
    url: '/toolsUserRelation/deleteUserRelation',
    method: 'delete',
    params
  })
}

export const getUserIdsByEnvironment = (params) => {
  return service({
    url: '/toolsUserRelation/getUserIdsByEnvironment',
    method: 'get',
    params
  })
}

File: server/plugin/tools/router/user_relation.go
package router

import (
	"github.com/gin-gonic/gin"
)

type userRelation struct{}

func (r *userRelation) Init(public, private *gin.RouterGroup) {
	group := private.Group("toolsUserRelation")
	group.POST("createUserRelation", apiInfo.UserRelation.CreateUserRelation)
	group.GET("findUserRelation", apiInfo.UserRelation.FindUserRelation)
	group.GET("getUserRelationList", apiInfo.UserRelation.GetUserRelationList)
	group.DELETE("deleteUserRelation", apiInfo.UserRelation.DeleteUserRelation)
	group.GET("getUserIdsByEnvironment", apiInfo.UserRelation.GetUserIdsByEnvironment)
}
